san/impl/missing_return.cpp

```diff
@@ -0,0 +1,137 @@
+#include "ast.hpp"
+#include "fmt/core.h"
+#include "san.hpp"
+#include <memory>
+#include <cassert>
+
+using namespace ast;
+
+namespace san {
+
+void MissingReturn::visit(Program* node) {
+    for (auto routine : node->routines) {
+        // Routines that don't return anything, can return nothing.
+        if (routine->returnType == nullptr) {
+            continue;
+        }
+
+        m_hasReturn = false;
+
+        routine->accept(*this);
+
+        if (!m_hasReturn) {
+            error(routine->begin, "missing a return statement on some execution paths");
+        }
```

Comment:
Maybe it is better to put this in the end of the violating `Body` to make it easier to highlight which branch exaclty is not returning

Reply:
But bodies can skip returns, and will be fine:
```
routine foo() : integer is
    if false then
        var bar is 42
    end
    return 42
end
```
If we'll move that check to the Bodies, it will surely cluster the code with additional checks.